test/suites/depend_different_headers.bash

```diff
@@ -0,0 +1,178 @@
+
+    # Compile dir3
+    # dir3 header change does not change object file compared to dir1, but ccache still adds
+    # additional .o/.d file in the cache due to different contents of the header file
+    cd ${BASEDIR3}
+    generate_reference_compiler_output
+    CCACHE_DEPEND=1 CCACHE_BASEDIR=${BASEDIR3} $CCACHE_COMPILE $DEPFLAGS -c test.c
+    expect_equal_object_files reference_test.o test.o
+    expect_equal_files reference_test.d test.d
+    expect_stat 'cache hit (direct)' 1
+    expect_stat 'cache hit (preprocessed)' 0
+    expect_stat 'cache miss' 3
+    expect_stat 'files in cache' 7      # 3x .o, 3x .d, 1x manifest
+
+    # Compile dir4
+    # dir4 header adds a new dependency
+    cd ${BASEDIR4}
+    generate_reference_compiler_output
+    CCACHE_DEPEND=1 CCACHE_BASEDIR=${BASEDIR4} $CCACHE_COMPILE $DEPFLAGS -c test.c
+    expect_equal_object_files reference_test.o test.o
+    expect_equal_files reference_test.d test.d
+    expect_different_files reference_test.d ${BASEDIR1}/test.d
+    expect_stat 'cache hit (direct)' 1
+    expect_stat 'cache hit (preprocessed)' 0
+    expect_stat 'cache miss' 4
+    expect_stat 'files in cache' 9      # 4x .o, 4x .d, 1x manifest
+
+    # Recompile dir1 - 2nd time
+    cd ${BASEDIR1}
+    generate_reference_compiler_output
+    CCACHE_DEPEND=1 CCACHE_BASEDIR=${BASEDIR1} $CCACHE_COMPILE $DEPFLAGS -c test.c
+    expect_equal_object_files reference_test.o test.o
+    expect_equal_files reference_test.d test.d
+    expect_stat 'cache hit (direct)' 2
+    expect_stat 'cache hit (preprocessed)' 0
+    expect_stat 'cache miss' 4
+    expect_stat 'files in cache' 9
+
+    # Recompile dir2
+    cd ${BASEDIR2}
+    generate_reference_compiler_output
+    CCACHE_DEPEND=1 CCACHE_BASEDIR=${BASEDIR2} $CCACHE_COMPILE $DEPFLAGS -c test.c
+    expect_equal_object_files test.o test.o
+    expect_equal_files reference_test.d test.d
+    expect_stat 'cache hit (direct)' 3
+    expect_stat 'cache hit (preprocessed)' 0
+    expect_stat 'cache miss' 4
+    expect_stat 'files in cache' 9
+
+    # Recompile dir3
+    cd ${BASEDIR3}
+    generate_reference_compiler_output
+    CCACHE_DEPEND=1 CCACHE_BASEDIR=${BASEDIR3} $CCACHE_COMPILE $DEPFLAGS -c test.c
+    expect_equal_object_files reference_test.o test.o
+    expect_equal_files reference_test.d test.d
+    expect_stat 'cache hit (direct)' 4
+    expect_stat 'cache hit (preprocessed)' 0
+    expect_stat 'cache miss' 4
+    expect_stat 'files in cache' 9
+
+    # Recompile dir4
+    cd ${BASEDIR4}
+    generate_reference_compiler_output
+    CCACHE_DEPEND=1 CCACHE_BASEDIR=${BASEDIR4} $CCACHE_COMPILE $DEPFLAGS -c test.c
+    expect_equal_object_files reference_test.o test.o
+    expect_equal_files reference_test.d test.d
+    expect_different_files reference_test.d ${BASEDIR1}/test.d
+    expect_stat 'cache hit (direct)' 5
+    expect_stat 'cache hit (preprocessed)' 0
+    expect_stat 'cache miss' 4
+    expect_stat 'files in cache' 9
+}
```

Comment:
[The last newline is missing](https://thoughtbot.com/blog/no-newline-at-end-of-file).